homeassistant/components/ios/__init__.py

```diff
@@ -240,6 +266,11 @@ async def async_setup(hass, config):
             )
         )
 
+    conf = config.get(DOMAIN)
+
+    if conf:
+        hass.http.register_view(iOSConfigView(conf))
```

Comment:
Should we register the view when setting up the entry instead, together with the other views?

Reply:
Under the modern/current architecture app installs are registered as entities under `mobile_app` not `ios`. I placed it here to ensure it is made available in the (valid and probably expected) case where no apps are registered with `ios`. I never actually quite understood why the push config view was registered in the `async_setup_entry` when it has no dependency on entity-specific data. I may be getting the wrong end on this all though.

Reply:
Setting up the entry is not directly connected to creating entities. The latter is done by the platforms. A config entry should be required for a working integration so it makes sense to register any services (in the broad sense) in the entry setup.

Reply:
@MartinHjelmare Thanks for that. I've made that change and double checked the _legacy_ view is still working, it is :)